### PR TITLE
Implements GoogleOAuth via Passport and Cookie based auth

### PR DESCRIPTION
**Pull request by lazycipher**
This PR implements Google OAuth and uses Cookie for authentication.
PR on backend: https://github.com/codeuino/social-platform-donut-backend/pull/197

### CODE REVIEW

**Comment by lazycipher** [comment]
@vaibhavdaren, @capedcrusader23, @Rupeshiya, @AuraOfDivinity 
Please review once.
Note: Code refactoring is left!

**Comment by lazycipher** [comment]
@vaibhavdaren, Both these need to be merged to work properly in deployments!
https://github.com/codeuino/social-platform-donut-backend/pull/197 (Merged)

File: src/actions/authAction.js [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
 import axios from 'axios';
 import { setAuthToken } from '../utils/setAuthToken';
-import jwt_decode from 'jwt-decode';
 import { errorHandler } from '../utils/errorHandler';
 import { setRequestStatus } from '../utils/setRequestStatus';
 import { BASE_URL } from './baseApi';
 import { customErrorHandler } from '../utils/customErrorHandler'
 let forgotPasswordToken = "";
 
+// load user
```
Comment (vaibhavdaren) [inline src/actions/authAction.js]:
```suggestion
```
its evident from the name

File: src/actions/authAction.js [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+      // update user role in localStorage
+      localStorage.setItem('admin', res.data.user.isAdmin)
+
+      // update user role in localStorage
+      localStorage.setItem('ticketModerator', res.data.user.isTicketsModerator)
+
+      // store orgId in localStorage
+      localStorage.setItem('orgId', res.data.user.orgId);
+
+      // if user is admin update admin in store
```
Comment (vaibhavdaren) [inline src/actions/authAction.js]: dry comments please improvise

File: src/actions/authAction.js [hunk 3]
```diff
@@ -1,13 +1,48 @@
...
+
+      // store orgId in localStorage
+      localStorage.setItem('orgId', res.data.user.orgId);
+
+      // if user is admin update admin in store 
+      dispatch({
+        type: SET_ADMIN,
+        payload: res.data.user.isAdmin
+      })
+      // window.location.href='/dashboard'
```
Comment (vaibhavdaren) [inline src/actions/authAction.js]: any particular reason to push this to prod?

File: src/actions/authAction.js [hunk 4]
```diff
@@ -1,13 +1,48 @@
...
+    }
+  } catch(error) {
+    console.log('user loading', error)
```
Comment (vaibhavdaren) [inline src/actions/authAction.js]: something like 
```suggestion
    console.log('error while loading user from localstorage ', error)
```

File: src/actions/authAction.js [hunk 5]
```diff
@@ -39,18 +74,11 @@ export const loginUser = (userInfo, history) => async (dispatch) => {
...
 
-      const token = res.data.token;
       dispatch(setRequestStatus(true));
-      
-      localStorage.setItem("jwtToken", (token));
-      setAuthToken(token);
-      
-      // update state with user
-      const decodedData = await jwt_decode(token);
       console.log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! PRINTING RES.DATA.USER !!!!!!!!!!!!!!1");
```
Comment (vaibhavdaren) [inline src/actions/authAction.js]:
```suggestion
      console.log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! PRINTING RES.DATA.USER !!!!!!!!!!!!!!1");
```

File: src/reducers/authReducer.js [hunk 6]
```diff
@@ -17,7 +17,7 @@ export default (state = initialState, action) => {
     case SET_CURRENT_USER: {
       return {
         ...state,
-        isAuthenticated: Boolean(typeof action.payload === 'object' && Object.keys(action.payload).length !== 0),
+        isAuthenticated: Boolean(action.payload.length !== 0 && action.payload!==undefined && typeof action.payload === 'string'),
```
Comment (vaibhavdaren) [inline src/reducers/authReducer.js:20]: cc @ksraj123

File: src/user/auth/login/login.js [hunk 7]
```diff
@@ -106,7 +106,7 @@ class Login extends Component {
               {Boolean(loginOptions?.github === true) ? (
                 <Col className="button-column" >
                 <a
-                  href="http://localhost:5000/auth/google"
+                  href="http://localhost:5000/user/auth/github"
```
Comment (vaibhavdaren) [inline src/user/auth/login/login.js]: chnage to compound url ---> base + endpoint

**Comment by vaibhavdaren** [comment]
> @vaibhavdaren, Both these need to be merged to work properly in deployments!
> [codeuino/social-platform-donut-backend#197](https://github.com/codeuino/social-platform-donut-backend/pull/197) (Merged)

Please fix these and get this peer reviewed

**Comment by lazycipher** [comment]
> Please fix these and get this peer reviewed

Sure thing! I'll do it asap!

File: src/reducers/authReducer.js (hunk 6 above)
Comment (lazycipher) [reply src/reducers/authReducer.js:20]: @ksraj123, what change needed here?

File: src/actions/authAction.js (hunk 5 above)
Comment (lazycipher) [reply src/actions/authAction.js]: removed! this console log.

File: src/index.js [hunk 8]
```diff
@@ -3,11 +3,14 @@ import ReactDOM from "react-dom";
 import "./index.css";
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
+import axios from 'axios';
 
 if (process.env.NODE_ENV !== "development") {
   console.log = () => {};
 }
 
+axios.defaults.withCredentials = true;
```
Comment (Rupeshiya) [inline src/index.js]: @lazycipher I think this should be only when user is authenticated.
Add this setAuthToken function in 
```
/utils/setAuthToken.js
axios.defaults.withCredentials = true;
```

**Review by vaibhavdaren — APPROVED** [review]
@jaskiratsingh2000 please work through the estimate and timelines of the project again.

**Pull request merged**